R/calcTimeSeriesStats.R

```diff
@@ -0,0 +1,65 @@
+#' Calculate Statistics of Time Series
+#'
+#' Take a quitte object and calculate given summary statistics of the time
+#' series of given variables in the quitte object. Return the result for each
+#' model, scenario, region, variable, and statistic in a data frame.
+#'
+#' @param data A quitte object.
+#' @param var character(n). The names of the variables to calculate the
+#'   statistics on.
+#' @param stats A named list of functions with two arguments. Each function will
+#'   be called with the vector of values as the first argument and the vector of
+#'   corresponding periods as the second argument.
+#' @param from,to Optional variables to filter the time series by period.
+#' @return A data frame with columns model, scenario, region, variable, value,
+#'   statistic, unit. "value" contains the calculated statistics. "statistic" is
+#'   name of the statistics, i.e., the name of the function in the argument
+#'   \code{stats}. The "unit" is the unit of the original variable and not
+#'   necessarily the correct unit for the calculated statistics.
+#' @examples
+#'   \dontrun{
+#'     calcTimeSeriesStats(
+#'       data, \
+#'       "Emi|CO2",
+#'       stats = list("Net 0 Period" = \(v, p) p[which(v <= 0)[1]]))
+#'   }
+#' @importFrom dplyr all_of %>% filter distinct group_by summarize mutate
+#'   left_join bind_rows
+#' @importFrom rlang .env .data
+#' @export
+calcTimeSeriesStats <- function(
+    data,
+    var,
+    stats,
+    from = -Inf,
+    to = 2100) {
+
+  stopifnot(quitte::is.quitte(data))
+  stopifnot(is.list(stats))
+  stopifnot(length(stats) > 0)
+  stopifnot(!is.null(names(stats)))
+  stopifnot(is.character(var))
+  stopifnot(is.numeric(from) && is.numeric(to))
+  stopifnot(length(from) == 1 && length(to) == 1)
+
+  idCols <- c("model", "scenario", "region", "variable")
+  data %>%
+    filter(.data$variable %in% .env$var, .data$period >= .env$from, .data$period <= .env$to) ->
```

Comment:
The linter should flag `->` as undesirable operator, please always use `<-` even when using pipes.

Reply:
true! I replaced the assignment operators accordingly